src/main/java/org/testng/TestNG.java

```diff
@@ -275,23 +275,28 @@ public void initializeSuitesAndJarFile() {
 
     m_isInitialized = true;
     if (m_suites.size() > 0) {
-    	//to parse the suite files (<suite-file>), if any
-    	for (XmlSuite s: m_suites) {
-        for (String suiteFile : s.getSuiteFiles()) {
-            Path rootPath = Paths.get(s.getFileName()).getParent();
-            try {
-                Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
-                for (XmlSuite cSuite : childSuites){
-                    cSuite.setParentSuite(s);
-                    s.getChildSuites().add(cSuite);
-                }
-            } catch (ParserConfigurationException | IOException | SAXException e) {
-                e.printStackTrace(System.out);
-            }
-        }
-
-    	}
-      return;
+	    //to parse the suite files (<suite-file>), if any
+		for (XmlSuite s: m_suites) {
+		   for (String suiteFile : s.getSuiteFiles()) {
+	        	Path suitePath = Paths.get(s.getFileName());
+	        	Path rootPath = suitePath.getParent();
+	        	try {
+	        		Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
+	        		for (XmlSuite cSuite : childSuites){
+	        			Path childSuite = Paths.get(cSuite.getFileName()).normalize();
+	        			Path parentSuite = Paths.get(suiteFile).normalize();
+	        			if(!childSuite.getFileName().equals(parentSuite.getFileName()))
```

Comment:
Maybe you are right but nobody is supposed to use the `Parser` class. 
And I still think the `parse()` contract has no sense and should parse the current suite, its children and add them instead.

BTW, at least, I suggere to add a new method that will parse children only. Currently, even the variable name is wrong: `childSuites`.
I think it is better to have this logic into `Parser` instead of `TestNG`.